plugins/ltac2/tac2core.ml

```diff
@@ -728,6 +728,11 @@ let () = define3 "constr_in_context" ident constr closure begin fun id t c ->
       Tacticals.tclZEROMSG (str "Variable already exists")
     else
       let open Context.Named.Declaration in
+      let sigma =
+        let t_ty = Retyping.get_type_of env sigma t in
+        let sigma, _ = Typing.type_judgment env sigma {uj_val=t; uj_type=t_ty} in
```

Comment:
please add a comment about why this is needed. t was just inferred to have type t_ty...

Reply:
We want t to be a type.

Reply:
also while we're doing this let's use the correct relevance

Reply:
fine, but please add a comment (in the sources).

Reply:
commented